src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStatic.cs

```diff
@@ -202,8 +202,8 @@ void OnSymbolEnd(SymbolAnalysisContext symbolEndContext)
                 return false;
             }
 
-            // Do not analyze constructors and finalizers.
-            if (methodSymbol.IsConstructor() || methodSymbol.IsFinalizer())
+            // Do not analyze constructors, finalizers, and indexers.
+            if (methodSymbol.IsConstructor() || methodSymbol.IsFinalizer() || methodSymbol.IsIndexer)
```

Comment:
I should be working against property symbol, not method symbol. Will close and open a new PR when I give another look.